internal/kibana/reconciler.go

```diff
@@ -228,10 +241,16 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaDeployment(proxyConfig
 		return kverrors.Wrap(err, "Failed to get oauth-proxy image")
 	}
 
+	oauth, err := getOAuthConfig(clusterRequest.client)
+	if err != nil {
+		return kverrors.Wrap(err, "Failed to get oauth config")
+	}
+
 	kibanaPodSpec := newKibanaPodSpec(
 		clusterRequest,
 		fmt.Sprintf("%s.%s.svc", clusterName, clusterRequest.cluster.Namespace),
 		proxyConfig,
+		oauth,
```

Comment:
Would it make sense to only push down the information we actually need (the expiry time) instead of the whole struct?

Reply:
oauth is actually a pointer so we're just passing the ref

Reply:
I know, this was not meant as a "performance enhancement".

Our code is only interested in one single information, but currently we push that down together with a lot of information we do not need. If we reduce this to just the one thing we need, we can simplify the code "further down". This usually also makes testing code more concise, because you do not need to provide testing data for the whole struct but just the one attribute you are interested in.

And because you pull the "parse and validate the value we need" code up, if there is an error it does not get generated by something way down in the details but on a much higher layer.

Reply:
I see your point